File: solutions/r/collatz-conjecture/1/collatz-conjecture.R
collatz_step_counter <- function(n) {
  if(!is.numeric(n)||any(n%%1!=0)||any(n < 1)) {
    stop('Error: Invalid Number')
  }
  steps <- function(k) {
    m <- 0L
    while(k != 1L) {
      m <- m + 1L
      k <- if(k%%2L==0L) k%/%2L else k*3L + 1L
    }
    return(m)
  }
  vapply(n,steps,integer(1))
}

File: solutions/julia/collatz-conjecture/1/collatz-conjecture.jl
function collatz_steps(n)
    n < 1 && throw(DomainError("Not a positive integer"))
    n == 1 && return 0
    steps = 0
    while n != 1
        steps += 1
        mod(n,2) == 1 ? n = n*3 + 1 : n = n/2
    end
    return steps
end
